packages/workflow/src/workflow.ts

```diff
@@ -526,3 +526,66 @@ export function uuid4(): string {
     4
   )}-${ho(view.getUint32(10), 8)}${ho(view.getUint16(14), 4)}`;
 }
+
+/**
+ * Patch or upgrade workflow code by checkin or stating that this workflow has a certain patch.
+ *
+ * TODO: See docs page for info
+ *
+ * If the workflow is replaying an existing history, then this function returns true if that
+ * history was produced by a worker which also had a `patched` call with the same `patchId`.
+ * If the history was produced by a worker *without* such a call, then it will return false.
+ *
+ * If the workflow is not currently replaying, then this call *always* returns true.
+ *
+ * Your workflow code should run the "new" code if this returns true, if it returns false, you
+ * should run the "old" code. By doing this, you can maintain determinism.
+ *
+ * @param patchId An identifier that should be unique to this patch. It is OK to use multiple
+ * calls with the same ID, which means all such calls will always return the same value.
+ */
+export function patched(patchId: string): boolean {
+  return patchInternal(patchId, false);
+}
+
+/**
+ * Indicate that a patch is being phased out.
+ *
+ * // TODO: See docs page, how to run queries, etc.
+ *
+ * Workflows with this call may be deployed alongside workflows with a {@link patched} call, but
+ * they must *not* be deployed while any workers still exist running old code without a
+ * {@link patched} call, or any runs with histories produced by such workers exist. If either kind
+ * of worker encounters a history produced by the other, their behavior is undefined.
+ *
+ * Once all live workflow runs have been produced by workers with this call, you can deploy workers
+ * which are free of either kind of patch call for this ID. Workers with and without this call
+ * may coexist, as long as they are both running the "new" code.
+ *
+ * @param patchId An identifier that should be unique to this patch. It is OK to use multiple
+ * calls with the same ID, which means all such calls will always return the same value.
+ */
+export function deprecatePatch(patchId: string): void {
+  patchInternal(patchId, true);
+}
+
+function patchInternal(changeId: string, deprecated: boolean): boolean {
+  // Always send the command to core, it's smart enough to ignore it if it was pointless.
+  // TODO: Interceptors? Do we care?
```

Comment:
IMO we don't need interceptors here, let's change the comment to say:
```suggestion
  // Patch operation does not support interception at the moment, if it did, this would be the place to start the interception chain
```